pkg/resources/instance/resource.go

```diff
@@ -484,6 +540,68 @@ func rUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag
 		}
 	}
 
+	// Attach/detach Block Storage Volumes
+	if d.HasChange(AttrBlockStorageVolumeIDs) {
+		o, n := d.GetChange(AttrBlockStorageVolumeIDs)
+		old := o.(*schema.Set)
+		cur := n.(*schema.Set)
+
+		if added := cur.Difference(old); added.Len() > 0 {
+			for _, id := range added.List() {
+				iid, err := egoscaleV3.ParseUUID(*instance.ID)
+				if err != nil {
+					return diag.Errorf("unable to parse instance ID: %s", err)
+				}
+				bid, err := egoscaleV3.ParseUUID(id.(string))
+				if err != nil {
+					return diag.Errorf("unable to parse block storage ID: %s", err)
+				}
+
+				request := egoscaleV3.AttachBlockStorageVolumeToInstanceRequest{
+					Instance: &egoscaleV3.InstanceTarget{
+						ID: iid,
+					},
+				}
+
+				op, err := clientV3.AttachBlockStorageVolumeToInstance(
+					ctx,
+					bid,
+					request,
+				)
+				if err != nil {
+					return diag.Errorf("unable to parse attached instance ID: %s", err)
+				}
+
+				_, err = clientV3.Wait(ctx, op, egoscaleV3.OperationStateSuccess)
+				if err != nil {
+					return diag.Errorf("failed to attach block storage: %s", err)
+				}
```

Comment:
there might be some shared code between this section and the rCreate function that could be reused from a function.

Reply:
> It makes sense, but i don't think it's worth the effort to refactor this resource as we are going to migrate the whole it to framework soon(ish).